spatialos/UnrealGDK
# * Integrated the Development Authentication Flow

**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This change enables the cloud test flow by integrating the Development Authentication Flow. Rather than hard-coding the dev-auth-token, you can specify it in the settings. Also, you can choose which deployment to automatically connect.

~~This change also enables connecting local UnrealServer worker to the cloud using the same auth flow as the client worker.~~

#### Release note
- Enabled new Development Authentication Flow

#### Tests
1. Build the worker assemblies and upload them to the cloud
2. Launch the cloud deployment with launch config "manual_worker_connection_only" = false
3. Generate Development Authentication Token with 'spatial project auth dev-auth-token'
4. Fill the token in the SpatialOS Runtime Settings, and enable 'Use Development Authentication Flow'
5. Start the client in the editor. Observe the inspector of the deployment that the client worker connects
6. ~~To enable connecting local UnrealServer worker to the cloud, check on 'Connect Local Server Worker' in the Settings mentioned above. Re-run step 5 and observe that both client and server worker connect to the deployment.~~

#### Documentation
WIP

#### Primary reviewers
@jessicafalk @Vatyx
